en/lessons/basics/collections.md

```diff
@@ -3,138 +3,363 @@ version: 1.3.1
+Tuples are the first kind of collection you are introduced to in Haskell. This
+is because they are simple, primitive, and have a terse, flexible, built in
+syntax. Tuples are sometimes referred to as anonymous records, but instead of
+referencing fields by name you reference them by position in the structure.
+Tuples can contain an arbitrary number of items, well this is not entirely
+true, the Haskell Report only requires up to 15-tuples and GHC comes with
+support for 62-tuples, but usually that is enough.We will focus on the 2-tuple
+since Haskell has good built in support for them. However, if you are
+interested this is what an 8-tuple looks like
+
+```haskell
+ghci> :t ('0', '1', '2', '3', '4', '5', '6', "8-tuple")
+('0', '1', '2', '3', '4', '5', '6', "8-tuple")
+  :: (Char, Char, Char, Char, Char, Char, Char, [Char])
+```
+__Note__: `:t` in the ghci repl shows the type of the value, it will print out a
+message of the form `term :: Type`
+
+### When to Use
+
+Tuples are not very flexible because they have a fixed length. This also means
+we have a lot of information about the structure of our data and therefore
+access is very perfomant `O(1)`. From a type safety perspective tuples are
+useful for associating data; saying "I required both of these things". For
+example if we wanted to represent a money transfer it wouldn't make sense to
+have just the amount or just the destination, we would want to group them `(Int,
+AccountId)`.
+
+There is an interesting parallel between functions that require n arguments and
+functions that require n-tuples. While "curried" functions are arguably more
+useful due to partial application, these two modalities are equivalent.
+
+```haskell
+ghci> :t (&&)
+(&&) :: Bool -> Bool -> Bool
+ghci> :t uncurry (&&)
+uncurry (&&) :: (Bool, Bool) -> Bool
+ghci> :t curry (uncurry (&&))
+curry (uncurry (&&)) :: Bool -> Bool -> Bool
+```
 
-```elixir
-iex> [3.14, :pie, "Apple"]
-[3.14, :pie, "Apple"]
+### Constructing Tuples
+
+We can construct tuples using a familiar syntax `(a,b)`.
+
+```haskell
+ghci> myTuple = (True,"hello")
+ghci> :t myTuple
+myTuple :: (Bool, [Char])
 ```
 
-Elixir implements list collections as linked lists.
-This means that accessing the list length is an operation that will run in linear time (`O(n)`).
-For this reason, it is typically faster to prepend than to append:
+### Consuming Tuples
 
-```elixir
-iex> list = [3.14, :pie, "Apple"]
-[3.14, :pie, "Apple"]
-# Prepending (fast)
-iex> ["π" | list]
-["π", 3.14, :pie, "Apple"]
-# Appending (slow)
-iex> list ++ ["Cherry"]
-[3.14, :pie, "Apple", "Cherry"]
+We can consume tuples by breaking them apart. With a 2-tuple the options are
+pretty simple. They have the predictable names `fst` and `snd`, and are
+available in the standard library, so no need to import anything!
+
+```haskell
+ghci> fst myTuple
+True
+ghci> snd myTuple
+"hello"
 ```
 
-### List Concatenation
+The types of these functions simple, yet reveal exactly how they behave. `fst`
```

Comment:
are simple

Reply:
@myShogoth for trivial changes it may make sense to use GitHub's "suggestions" mechanism: just as much typing for you, and one-click approval for the author...